test_conformance/extensions/cl_khr_command_buffer/command_buffer_get_command_buffer_info.cpp

```diff
@@ -0,0 +1,334 @@
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+#define test_expected_info(cond)                                               \
+    {                                                                          \
+        if (cond)                                                              \
+        {                                                                      \
+            log_error("clGetCommandBufferInfoKHR return values not as "        \
+                      "expected\n");                                           \
+            return TEST_FAIL;                                                  \
+        }                                                                      \
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RunQueuesInfoTest()
+    {
+        cl_int error = TEST_PASS;
+
+        // record command buffers
+        error = RecordCommandBuffer();
+        test_error(error, "RecordCommandBuffer failed");
+
+        // vector containter added due to potential future growth, at the moment
+        // spec of cl_khr_command_buffer says command-buffer accepts only 1
+        // queue
+        std::vector<cl_command_queue> expect_queue_list = { queue };
+        cl_uint num_queues = 0;
+        size_t ret_value_size = 0;
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_NUM_QUEUES_KHR, sizeof(cl_uint),
+            &num_queues, &ret_value_size);
+        test_error(error, "clGetCommandBufferInfoKHR failed");
+
+        test_expected_info(ret_value_size > sizeof(cl_int));
+
+        test_expected_info(num_queues != expect_queue_list.size());
+
+        std::vector<cl_command_queue> ql(num_queues);
+        size_t expect_size = ql.size() * sizeof(cl_command_queue);
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_QUEUES_KHR, expect_size,
+            &ql.front(), &ret_value_size);
+        test_error(error, "clGetCommandBufferInfoKHR failed");
+
+        test_expected_info(ret_value_size > expect_size);
+
+        // We can not check if this is the right queue because this is an opaque
+        // object, test against NULL.
+        for (int i = 0; i < ql.size(); i++)
+        {
+            test_expected_info(ql[i] == NULL);
+        }
+        return TEST_PASS;
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RunRefCountInfoTest()
+    {
+        cl_int error = CL_SUCCESS;
+
+        // record command buffer
+        error = RecordCommandBuffer();
+        test_error(error, "RecordCommandBuffer failed");
+
+        // collect initial reference count
+        cl_uint init_ref_count = 0;
+        size_t ret_value_size = 0;
```

Comment:
Don't think we ever check the returned `ret_value_size` in this test

Reply:
Done